dbt/tests/generic/test_res_class_matches_pardat.sql

```diff
@@ -0,0 +1,62 @@
+-- For all residential parcels in a given model, test that there is at least one
+-- class code that matches a class code for that parcel in pardat.
+--
+-- By default, the test will compare the first 3 digits of each set of classes;
+-- if `major_class_only=true`, however, the test will compare the first digit
+-- only.
+--
+-- The test filters for residential parcels by anti-joining the model against
+-- `comdat`. This join will be performed using (parid, taxyr, card) by default,
+-- but the join can be configured to use (parid, taxyr) instead by setting
+-- `filter_for_res_by_card=false`.
+{% test res_class_matches_pardat(
+    model,
+    column_name,
+    filter_for_res_by_card=true,
+    major_class_only=false,
+    additional_select_columns=[]
+) %}
+
+    {%- set num_class_digits = 1 if major_class_only else 3 %}
```

Comment:
Note that while the original `asmt_all` test compared the first 3 digits of the class codes, the original `dweldat` test didn't (`oby` and `land` both look at major classes). I'm assuming it's fine to perform the `dweldat` test using the first 3 digits of each class for simplicity, but I could be wrong!